### PR TITLE
feat: quiz | show answers

### LINKED ISSUE #606: [FEATURE] Quiz section : add answers
### Detailed description

when we take the quiz , if our answer is right it will skip to the next question ,if the answer is wrong it will show us the right answer (the right answer will be highlighted in green and the wrong one in red ) 

### Context

 Users can benefit from a more educational experience by understanding which questions they answered correctly and learning from those they got wrong. in addition to increasing user engagement. 

### Possible implementation

_No response_

### Additional information

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by nassim-ayadi**
## Fixes Issue

 Closes #606

## Changes proposed

-if the user gets the answer wrong his answer will be highlighted in red and the correct answer will be highlighted in green 
-to skip to the next question the user must click on the right answer (the green one )

## Screenshots

Capture d'écran 2024-01-20 002919
Capture d'écran 2024-01-20 002945

## Code of Conduct

- [X] By submitting this pull request, I confirm I've read and complied with the [CoC](https://github.com/thecyberworld/Support/blob/main/CODE_OF_CONDUCT.md) 🖖

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

---

You can also join our [Discord](https://discord.gg/QHBPq6xP5p) community.
Feel free to check out other cool repositories of the [Thecyberworld](https://github.com/thecyberworld).
Join the Thecyberworld GitHub Organisation by raising an [issue](https://github.com/thecyberworld/Support/issues/new?assignees=&labels=invite+me+to+the+organisation&template=invitation.yml&title=Please+invite+me+to+the+GitHub+Community+Organization) (you will be sent an invitation).

### CODE REVIEW

**Comment by kabir0x23** [comment]
Hi thanks for contributing, will check the pr shortly, thank you

**Comment by ArkadiK94** [comment]
> -to skip to the next question the user must click on the right answer (the green one )

How will the user know that he needs to click the right answer to skip to next question?
I see two options:
- make an small animation of green flickering on the right answer. 
- skip to next question after few seconds automatically without the need of the user clicking the right answer.

**Comment by nassim-ayadi** [comment]
i think making a small animation on the right answer is the best choice because i cannot set standard timer for everyone (some are fest readers some are slow )

**Comment by kabir0x23** [comment]
are we showing correct/wrong ans at same time?
 
 Wwe can do one thing in the end
 
 below the scores, we can map all of the questions with correct / wrong ans user did, instead of giving at first place

**Comment by nassim-ayadi** [comment]
that would be the optimal way for a standard quiz.
but i think that the quizzes in this website are fast and  short (i think they are the best way to get the user involved ) .
 so if we map all the correct/wrong answers at the end with the score it will make the prosses longer : to read the question again and read the correct answer for every question the user got wrong .

**Comment by ArkadiK94** [comment]
@kabir0x23 What do you think? 
In my opinion, @nassim-ayadi idea is great.

**Comment by kabir0x23** [comment]
> @kabir0x23 What do you think? In my opinion, @nassim-ayadi idea is great.

if we are showing this ans is wrong,we should give a user some to to read and understad the ans properly

if it shows this is worng and it will move to next stage under 5 sec or instant, then user won't able to understand why it was wrong or correct

heshe have to come back and see what was the correct one, if there are 5-10 quiz it will hard to mange which one was correct

**Comment by ArkadiK94** [comment]
@kabir0x23 we can add "next" button, which will be enabled only after the user picks an answer.

**Comment by kabir0x23** [comment]
> @kabir0x23 we can add "next" button, which will be enabled only after the user picks an answer.

this will work

**Comment by nassim-ayadi** [comment]
Great idea i will work on this

**Comment by nassim-ayadi** [comment]
Capture d'écran 2024-01-29 204430
i have created the skip button and tested it , can someone please  verify the code

**Comment by ArkadiK94** [comment]
@nassim-ayadi There is a strange behavior when I hover an answer. Pls, make it to be as before.

https://github.com/thecyberworld/TheCyberHUB/assets/76536506/3a055292-414d-4f53-aeca-efd1ab14bc9a

**Comment by ArkadiK94** [comment]
In addition, when the answer is right, I expect the right answer to be green for 2 seconds and move to next question automatically. In my opinion, in this way the users will know that they answer correctly on this question.

**Comment by nassim-ayadi** [comment]
Noted. working on it

**Comment by kabir0x23** [comment]
> In addition, when the answer is right, I expect the right answer to be green for 2 seconds and move to next question automatically. In my opinion, in this way the users will know that they answer correctly on this question.

nice 👌

**Comment by nassim-ayadi** [comment]
@ArkadiK94 i implemented the changes we discussed .can you please check them  out ? (is 2 second s enough to skip question ? ) .

**Comment by ArkadiK94** [comment]
@nassim-ayadi the time of skipping question is ok, but there is a small glitch. When the user picks the right answer, it skips to next question but the place of the right answer in previous question get green for a sec in this question.



https://github.com/thecyberworld/TheCyberHUB/assets/76536506/cc75c0fa-a9d1-4884-9971-5e62b07346dd

File: src/components/Resources/Quiz/Categories/QuizPage.jsx [hunk 1]
```diff
@@ -33,6 +34,10 @@ export default function QuizPage() {
     const [currentQuestion, setCurrentQuestion] = useState(0);
     const [score, setScore] = useState(0);
     const [showScore, setShowScore] = useState(false);
+    const [showAnswer, setshowAnswer] = useState(false);
```
Comment (ArkadiK94) [inline src/components/Resources/Quiz/Categories/QuizPage.jsx:37]: ```setShowAnswer```
typo

**Review by ArkadiK94 — APPROVED** [review]
Great work, please, only fix the glitch and we could merge it :)

**Comment by ArkadiK94** [comment]
@kabir0x23 There is a small glitch. Would you like to merge it anyway and open a new issue about it?

**Comment by nassim-ayadi** [comment]
Ok no problem i ll fix it

**Comment by nassim-ayadi** [comment]
> @nassim-ayadi the time of skipping question is ok, but there is a small glitch. When the user picks the right answer, it skips to next question but the place of the right answer in previous question get green for a sec in this question.
> 
>  c4c9f6ab-9275-433d-b998-fad1dc3cf378.mp4

i tried to reproduce the glitch i did not succeed . can you please point out where is the problem exactly ?

https://github.com/thecyberworld/TheCyberHUB/assets/150674824/0d85abf2-b9ef-41b6-98f3-97b863cc9fac

**Comment by nassim-ayadi** [comment]
> @nassim-ayadi Look closer , I see it in the video that you uploaded , when it moves to next question for a second you see one of the answers in green. You also can make the network slower so you will have more time to see it.

Yes i played in slow motion , i noticed it , i ll fix it

**Comment by nassim-ayadi** [comment]
@ArkadiK94 I've set the transition to "none" , i think i fixed the glitch or minimized it (i am not sure ), can you please check it out

**Review by ArkadiK94 — APPROVED** [review]

**Comment by ArkadiK94** [comment]
@kabir0x23 What do you think? let's leave it in this way?

**Comment by ArkadiK94** [comment]
@nassim-ayadi Great work, @kabir0x23 said that this way is preferred one so I will merge this pr.

**Pull request merged**